src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -158,6 +158,11 @@ object ScalafixPlugin extends AutoPlugin {
         SettingKey[Boolean]("bspEnabled") := false
       )
     ),
+    scalafixInterfaceProvider := ScalafixInterface.fromToolClasspath(
```

Comment:
I didn't check but this will likely result in a significant memory increase as a separate instance of scalafix (together with the scala standard lib) will be classloaded for **each** project. Memoizing the provider per scalafixScalaBinaryVersion should do the trick, but I guess we will need to add some custom code as we can't rely on sbt scope to do that.

Reply:
If I'm reading the code right, there might be duplication in the event there isn't any custom tool paths, but we can correct for that by change the cache to be keyed on an `Option[ToolClasspath]` instead of just a `ToolClasspath` and we can create the default once and store it there.

Reply:
@bjaglin Would you like me to make that cache change?  Is there some other change I can make to get this accepted?

Reply:
> If I'm reading the code right, there might be duplication in the event there isn't any custom tool paths, but we can correct for that by change the cache to be keyed on an Option[ToolClasspath] instead of just a ToolClasspath and we can create the default once and store it there.

My concern is that `ScalafixInterface.fromToolClasspath()` is now called once per project instead of once per build, and there is no memoization inside. The fact that the key was at the build level guaranteed the singleton nature of the provider. The `ToolClasspath` [layering](https://github.com/bjaglin/sbt-scalafix/blob/568f1a2e435b225117aacae59391d505da9f1b6a/src/main/scala/scalafix/internal/sbt/ScalafixInterface.scala#L15-L31) is handled separately and won't be directly affected by this change.

> Would you like me to make that cache change?

Yes, I am afraid I can't get this as-is as it brings a serious regression. Although I am not sure we were talking about the same change, see my clarification above.

To illustrate my previous comment, I monitored the RAM usage of `scalafixAll --no-cache` on a fresh sbt session, against a warm build with many projects (scalafix). Note that this build uses sbt-projectmatrix and [only run scalafix on scala 2.13 projects](https://github.com/scalacenter/scalafix/blob/11a95ff17954518b0f28b0798654f7b1af0d4cd2/project/ScalafixBuild.scala#L306-L319) (for the same reason you are trying to address), but that's irrelevant here. I forced a GC a few seconds before I took the screenshot in both cases.

### sbt-scalafix 0.11.1+23-b71ae18a-SNAPSHOT (this PR)

![Screenshot from 2023-12-04 23-06-10](https://github.com/scalacenter/sbt-scalafix/assets/349077/5083c14a-5548-4ede-aff4-609820483b07)

```
[success] Total time: 72 s (01:12), completed Dec 4, 2023, 11:01:13 PM
```

### sbt-scalafix 0.11.1

![Screenshot from 2023-12-04 23-10-17](https://github.com/scalacenter/sbt-scalafix/assets/349077/35609fe5-cb1d-48c1-9e30-463420076d6e)

```
[success] Total time: 54 s, completed Dec 4, 2023, 11:08:16 PM
```

Reply:
@bjaglin Pushed an update that memoizes `fromToolClasspath`, let me know if this works for you.